Minor doc fixes (#5139)
PyTorchLightning/pytorch-lightning: docs/source/multi_gpu.rst

        z = torch.Tensor(2, 3)
        z = z.type_as(x)

The :class:`~pytorch_lightning.core.lightning.LightningModule` knows what device it is on. You can access the reference via ``self.device``.
Sometimes it is necessary to store tensors as module attributes. However, if they are not parameters they will
remain on the CPU even if the module gets moved to a new device. To prevent that and remain device agnostic,

> register the tensor as a buffer in your modules's `__init__` method with :meth:`~torch.nn.Module.register_buffer`.

Reviewer: ```suggestion
register the tensor as a buffer in your modules's ``__init__`` method with :meth:`~torch.nn.Module.register_buffer`.
```

Remove samplers
^^^^^^^^^^^^^^^

> In PyTorch, you must use `DistributedSampler <https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler>`_

Reviewer 2: You can try this, it should work:
```suggestion
In PyTorch, you must use :class:`~torch.utils.data.distributed.DistributedSampler`
```

        dataset = MNIST(...)
        return DataLoader(dataset)

.. note::
    By default it will add ``shuffle=True`` for train sampler and ``shuffle=False`` for val/test sampler.

>     ``drop_last`` in DistributedSampler will be set to its default value in PyTorch.

Reviewer 3: let's add ref also for `DistributedSampler`

Implement Your Own Distributed (DDP) training
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> If you need your own way to init PyTorch DDP you can override: :meth:`pytorch_lightning.plugins.ddp_plugin.DDPPlugin.init_ddp_connection`.

Reviewer 2: ```suggestion
If you need your own way to init PyTorch DDP you can override :meth:`~pytorch_lightning.plugins.ddp_plugin.DDPPlugin.init_ddp_connection`.
```
it's part of the sentence, so the colon should not be necessary? But I suggest to shorten the name with the tilde.
